### ISSUE TITLE
Windows doesn't support cudaMemPrefetchAsync()

### ISSUE THREAD

**Issue by stoperro** [TimDettmers/bitsandbytes#453]
Also memory oversubscription is not supported https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#system-requirements which I presume means paged optimizer that overcomes memory spikes won't work on windows.

This results in below error in QLoRA training:
```
rror invalid device ordinal at line 359 in file F:\Buildy\bitsandbytes_acpopescu\csrc\pythonInterface.c
C:\arrow\cpp\src\arrow\filesystem\s3fs.cc:2598:  arrow::fs::FinalizeS3 was not called even though S3 was initialized.  This could lead to a segmentation fault at exit
```

(note: above wasn't caused by old transformers version)

**Comment by stoperro** [comment]
Above linked solution that works for me - checking capabilities for that feature before running the call as hinted in https://stackoverflow.com/a/43430831/950131 . The call is fast, so maybe no need to cache the answer.

**Comment by jllllll** [comment]
Does this issue only apply to QLoRA training?

**Comment by TimDettmers** [comment]
This is a duplicate of #477, please redirect all discussion there. 

TL;DR: I need to think if I will support Maxwell or not. There might be a workaround for Maxwell support by excluding  Paged Optimziers.

**Comment by TimDettmers** [comment]
This has been fixed and pushed to pip. Memory problems might remain, but these are Windows-specific and there is nothing I can do about that. Thank you for the fix @stoperro , this was an important bugfix.
